src/tools/rbd_mirror/group_replayer/Replayer.cc

```diff
@@ -1325,6 +1381,66 @@ void Replayer<I>::finish_shut_down() {
   }
 }
 
+
+template <typename I>
+bool Replayer<I>::get_replay_status(std::string* description) {
+  dout(10) << dendl;
+
+  std::unique_lock locker{m_lock};
+  if (m_state != STATE_REPLAYING) {
+    locker.unlock();
+    derr << "replay not running" << dendl;
+    return false;
+  }
+
+  json_spirit::mObject root_obj;
+  root_obj["last_snapshot_complete_seconds"] = m_last_snapshot_complete_seconds;
+  root_obj["last_snapshot_bytes"] = m_last_snapshot_bytes;
+
+  auto remote_gp_snap_ptr = get_latest_mirror_group_snapshot(m_remote_group_snaps);
```

Comment:
Is it okay to use the in-memory list of remote group snaps?

Do I need to synchronously fetch the list of remote group snaps for the up-to-date list? This seems expensive